File: AutomationService/AutomationService/Features/EmailVersand/Domain/Services/EmailVersandErgebnis.cs
namespace AutomationService.Features.EmailVersand.Domain.Services;

/// <summary>
/// Was beim Versand herauskam. Der Zeitpunkt geht an den Vorgang zurück, damit
/// die Oberfläche den Abschluss begründen kann ("am … an 2 Empfänger gesendet")
/// — der Abschluss selbst bleibt der eigene Schritt des Anwalts (§4.8).
/// </summary>
/// <param name="GesendetAm">Zeitpunkt der Übergabe an den SMTP-Server.</param>
/// <param name="Empfaenger">Alle angeschriebenen Adressen (An und Kopie).</param>
/// <param name="ImGesendetOrdner">
/// True, wenn die Nachricht nachweislich im Ordner "Gesendet" des Postfachs
/// liegt — entweder vom Anbieter selbst abgelegt oder von hier nachgetragen.
/// §4.7 nennt diesen Ordner als einzigen Versandnachweis.
/// </param>
/// <param name="Hinweis">
/// Klartext zu einem Nebenbefund, der den Versand nicht verhindert hat (z. B.
/// die Kopie in "Gesendet" ließ sich nicht ablegen). Null im Regelfall.
/// </param>
public sealed record EmailVersandErgebnis(
    DateTimeOffset GesendetAm,
    IReadOnlyList<string> Empfaenger,
    bool ImGesendetOrdner,
    string? Hinweis);

File: Automation_App_Frontend/lib/features/email_versand/domain/entities/email_versand_ergebnis.dart
import 'package:equatable/equatable.dart';

/// Was der Versand ergeben hat (§4.7). Der Zeitpunkt begründet später das
/// vorbelegte Häkchen im Abschlussdialog — abgeschlossen wird der Vorgang
/// trotzdem von Hand (§4.8).
class EmailVersandErgebnis extends Equatable {
  final DateTime gesendetAm;
  final List<String> empfaenger;

  /// True, wenn die Nachricht nachweislich im Ordner „Gesendet" liegt — der
  /// einzige Versandnachweis, den §4.7 vorsieht.
  final bool imGesendetOrdner;

  /// Nebenbefund, der den Versand nicht verhindert hat; sonst null.
  final String? hinweis;

  const EmailVersandErgebnis({
    required this.gesendetAm,
    required this.empfaenger,
    required this.imGesendetOrdner,
    this.hinweis,
  });

  factory EmailVersandErgebnis.fromJson(Map<String, dynamic> json) {
    return EmailVersandErgebnis(
      gesendetAm:
          DateTime.tryParse(json['gesendetAm'] as String? ?? '')?.toLocal() ??
          DateTime.now(),
      empfaenger:
          (json['empfaenger'] as List?)?.whereType<String>().toList() ??
          const [],
      imGesendetOrdner: json['imGesendetOrdner'] as bool? ?? false,
      hinweis: json['hinweis'] as String?,
    );
  }

  @override
  List<Object?> get props => [
    gesendetAm,
    empfaenger,
    imGesendetOrdner,
    hinweis,
  ];
}
